src/main/kotlin/io/snyk/plugin/net/SnykApiClient.kt

```diff
@@ -29,13 +31,25 @@ class SnykApiClient(
             log.debug("Executing request to $apiName")
             val response = retrofitCall.execute()
             if (!response.isSuccessful) {
+                if (response.code() == 422) {
+                    val cliConfigSetting = Gson().fromJson<CliConfigSettings?>(response.errorBody()?.string(), CliConfigSettings::class.java)
+                    if (cliConfigSetting?.userMessage?.isNotEmpty() == true) {
+                        throw ClientException(cliConfigSetting.userMessage)
+                    }
+                    return null
+                }
                 log.warn("Failed to execute `$apiName` call: ${response.errorBody()?.string()}")
                 return executeRequest(apiName, retrofitCall.clone(), retryCounter - 1)
             } else {
                 return response.body()
             }
         } catch (t: Throwable) {
             log.warn("Failed to execute '$apiName' network request: ${t.message}", t)
+            if (t is ClientException) {
```

Comment:
You can just have a catch block before the block with t: Throwable instead of an if.